### DISCUSSION TITLE
Show how to convert from requirements.txt/Pipenv to Poetry

### DISCUSSION THREAD

**Discussion by rvanlaar** [python-poetry/poetry#5942] [Q&A]
- [X] I have searched the [issues](https://github.com/sdispater/poetry/issues) of this repo and believe that this is not a duplicate.

## Issue

I would like to use poetry. It took me waaaaay to long to find this issue:
https://github.com/sdispater/poetry/issues/663
It's closed but has nuggets with wisdom for how to import from other package managers.

Please, Please, include at least this issue in the documentation.

**Comment by rvanlaar** [comment]
There is a tool that supports migrating: https://dephell.org/docs/cmd-deps-convert.html

**Comment by ZeroAurora** [comment]
~~It would be better to have some commands like `poetry migrate ./requirements.txt` or `poetry migrate --pipenv ./Pipfile`. I don't want to install another tool.~~
Oh sorry I didn't read #663 .

**Comment by zupo** [comment]
Here's an example how to convert a "realworld app" from pipenv to poetry: https://github.com/niteoweb/pyramid-realworld-example-app/pull/84

**Comment by bolinocroustibat** [comment]
It should definitely be in the documentation, and maybe in the "get started" doc.

**Comment by FilBot3** [comment]
As a "work around" until some sort of capability comes about, you can try something like this: https://stackoverflow.com/a/62787881/2009612

```bash
cat requirements.txt | xargs poetry add
```

**Comment by sinoroc** [comment]
Use `dephell deps convert`: https://dephell.readthedocs.io/cmd-deps-convert.html

**Comment by dbuse** [comment]
There is also https://pypi.org/project/pipenv-poetry-migrate/

**Comment by rvanlaar** [comment]
> @rvanlaar error I get with dephell
> 
> ```bash
>  dephell deps convert --from-format=pipenv --to-format=poetr
> y
> usage: dephell deps convert [OPTIONS]
> dephell deps convert: error: argument --from-format: invalid choice: 'pipenv' (choose from 'conda', 'egginfo', 'flit', 'imports', 
> 
> 'installed', 'pip', 'pipfile', 'pipfilelock', 'piplock', 'poetry', 'poetrylock', 'pyproject', 'sdist', 'setuppy', 'wheel')
> ```

Have a look at example 5. It shows what you need. https://dephell.readthedocs.io/cmd-deps-convert.html#more-examples

**Comment by tombohub** [comment]
@rvanlaar  thanks, do you know why is it saying this now:

```
dephell deps convert --from-format=pipfilelock --from-path=Pipfile.lock --to-format=poetrylock --to-path=poetry.lock

WARNING cannot find tool.dephell section in the config (path=pyproject.toml)
ERROR InvalidSpecifier: ==not in3.0
```

how do I use pipenv.lock?

**Comment by rvanlaar** [comment]
I would ask for support at the dephell project. I'm a long time poetry user and haven't had to use the conversion tools in quite some time.

**Comment by tombohub** [comment]
@rvanlaar no problem thanks, the project dephell is archived now btw

**Comment by src-r-r** [comment]
I 2nd this. I like to use cookiecutter a lot, as well as poetry. Unfortunately a lot of the cookiecutter python projects are `requirements/setup.py`-based.

There is an `export` command, so would it seem reasonable to include a poetry `import` command that creates the project from a setup file (if it exists) and a list of `requirements.txt`s?

**Comment by src-r-r** [comment]
And I got a little too impatient, so I created this Python tool to do just this (called stanza...get it?). Parses requirements.txt files and automatically names and versions your project. It even allows for referenced requirements "e.g. `-r ...`"

https://github.com/src-r-r/python-stanza

I created it in a rush (but still tested it), so contributions welcome!

**Comment by N3X15** [comment]
I should note that dephell is now archived, domain taken over by squatters, and therefore dead. Other solutions should be used.
